src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1982,76 +1960,82 @@ namespace cryptonote
     return bce;
   }
   //-----------------------------------------------------------------------------------------------
-  bool core::handle_block_found(block& b, block_verification_context &bvc)
+  void core::on_synchronized()
+  {
+  }
+  //-----------------------------------------------------------------------------------------------
+  void core::safesyncmode(const bool onoff)
+  {
+    m_blockchain_storage.safesyncmode(onoff);
+  }
+  //-----------------------------------------------------------------------------------------------
+  bool core::add_new_block(const block& b, block_verification_context& bvc, checkpoint_t const *checkpoint)
+  {
+    bool result = m_blockchain_storage.add_new_block(b, bvc, checkpoint);
+    if (result)
+    {
+      // TODO(loki): PERF(loki): This causes perf problems in integration mode, so in real-time operation it may not be
+      // noticeable but could bubble up and cause slowness if the runtime variables align up undesiredly.
+      relay_service_node_votes(); // NOTE: nop if synchronising due to not accepting votes whilst syncing
+    }
+    return result;
+  }
+  //-----------------------------------------------------------------------------------------------
+  bool core::handle_block_found(block &b, block_verification_context &bvc)
   {
     bvc = {};
     std::vector<block_complete_entry> blocks;
-    m_miner.pause();
     {
-      LOKI_DEFER { m_miner.resume(); };
-      try
-      {
+      try {
         blocks.push_back(get_block_complete_entry(b, m_mempool));
       }
-      catch (const std::exception &e)
-      {
+      catch (const std::exception &e) {
         return false;
       }
       std::vector<block> pblocks;
-      if (!prepare_handle_incoming_blocks(blocks, pblocks))
-      {
+      if (!prepare_handle_incoming_blocks(blocks, pblocks)) {
         MERROR("Block found, but failed to prepare to add");
         return false;
       }
       add_new_block(b, bvc, nullptr /*checkpoint*/);
       cleanup_handle_incoming_blocks(true);
-      m_miner.on_block_chain_update();
```

Comment:
This call is quite important to update the block template otherwise the miner will keep trying to generate outdated blocks. I see that the miner interface has a get_block_template, but I don't think it's being used at the moment?